algorithms/kernel/dtrees/forest/regression/oneapi/df_regression_train_dense_default_oneapi_impl.i

```diff
@@ -0,0 +1,1245 @@
+                                                                                             UniversalBuffer & treeOrderBuf, size_t nRows,
+                                                                                             size_t nFeatures)
+{
+    DAAL_ITTNOTIFY_SCOPED_TASK(compute.doLevelPartition);
+
+    services::Status status;
+
+    auto & context = services::Environment::getInstance()->getDefaultExecutionContext();
+
+    auto & kernel = kernelDoLevelPartition;
+
+    {
+        KernelArguments args(5);
+        args.set(0, data, AccessModeIds::read);
+        args.set(1, nodeList, AccessModeIds::read);
+        args.set(2, treeOrder, AccessModeIds::read);
+        args.set(3, treeOrderBuf, AccessModeIds::write);
+        args.set(4, (int)nFeatures);
+
+        size_t localSize = _preferableSubGroup;
+
+        KernelRange local_range(localSize, 1);
+        KernelRange global_range(localSize, nNodes);
+
+        KernelNDRange range(2);
+        range.global(global_range, &status);
+        DAAL_CHECK_STATUS_VAR(status);
+        range.local(local_range, &status);
+        DAAL_CHECK_STATUS_VAR(status);
+
+        context.run(range, kernel, args, &status);
+        DAAL_CHECK_STATUS_VAR(status);
+    }
+
+    DAAL_CHECK_STATUS_VAR(partitionCopy(treeOrderBuf, treeOrder, 0, nRows));
+
+    return status;
+}
+
+template <typename algorithmFPType, decision_forest::regression::training::Method method>
+services::Status RegressionTrainBatchKernelOneAPI<algorithmFPType, method>::computeBestSplitByHistogram(
+    const UniversalBuffer & nodeHistogramList, UniversalBuffer & selectedFeatures, size_t nSelectedFeatures, UniversalBuffer & nodeList,
+    UniversalBuffer & nodeIndices, size_t nodeIndicesOffset, UniversalBuffer & binOffsets, UniversalBuffer & impList,
+    UniversalBuffer & nodeImpDecreaseList, bool updateImpDecreaseRequired, size_t nNodes, size_t nMaxBinsAmongFtrs, size_t minObservationsInLeafNode,
+    algorithmFPType impurityThreshold)
+{
+    DAAL_ITTNOTIFY_SCOPED_TASK(compute.computeBestSpitByHistogramLevel);
+
+    services::Status status;
+
+    auto & context = services::Environment::getInstance()->getDefaultExecutionContext();
+
+    auto & kernel = kernelComputeBestSplitByHistogram;
+
+    {
+        KernelArguments args(13);
+        args.set(0, nodeHistogramList, AccessModeIds::read);
+        args.set(1, selectedFeatures, AccessModeIds::read);
+        args.set(2, (int)nSelectedFeatures);
+        args.set(3, binOffsets, AccessModeIds::read);
+        args.set(4, nodeList, AccessModeIds::readwrite); // nodeList will be updated with split attributes
+        args.set(5, nodeIndices, AccessModeIds::read);
+        args.set(6, nodeIndicesOffset);
+        args.set(7, impList, AccessModeIds::write);
+        args.set(8, nodeImpDecreaseList, AccessModeIds::write);
+        args.set(9, (int)updateImpDecreaseRequired);
+        args.set(10, (int)nMaxBinsAmongFtrs);
+        args.set(11, (int)minObservationsInLeafNode);
```

Comment:
`int` is not standardized size in c++. In OpenCL  are standardized  - `int = int32` . For good, you need to replace int

Reply:
replaced